sympy/functions/elementary/trigonometric.py

```diff
@@ -2251,6 +2251,16 @@ def _eval_as_leading_term(self, x):
         else:
             return self.func(arg)
 
+    def _eval_is_real(self):
+        x = self.args[0]
+        if x.is_real == False:
+            return False
+        y = abs(x) - 1
```

Comment:
Maybe that does not help because the x in the function is not the same as that in the argument. Probably the condition  x.is_real == False is not sufficient as it does not preclude None. One should make sure that x.is_real is True after the test.